src/net/sourceforge/kolmafia/session/BanishManager.java

```diff
@@ -162,312 +228,233 @@ public static final void loadBanishedMonsters() {
-    } else if (banishName.equals("banishing shout") || banishName.equals("howl of the alpha")) {
-      String pref = monsterName;
-      String[] monsters = Preferences.getString("banishingShoutMonsters").split("\\|");
-      for (int i = 0; i < monsters.length && i < 2; ++i) {
-        if (monsters[i].length() > 0) {
-          pref += "|" + monsters[i];
+    switch (banisher) {
+      case NANORHINO:
+        Preferences.setString("_nanorhinoBanishedMonster", monster.getName());
+        break;
+      case BANISHING_SHOUT:
+      case HOWL_OF_THE_ALPHA:
+        {
+          Preferences.setString(
+              "banishingShoutMonsters",
+              Stream.concat(
+                      Stream.of(monster.getName()),
+                      Arrays.stream(Preferences.getString("banishingShoutMonsters").split("\\|"))
+                          .limit(2)
+                          .filter(Predicate.not(String::isEmpty)))
+                  .collect(Collectors.joining("|")));
+          break;
         }
-      }
-      Preferences.setString("banishingShoutMonsters", pref);
-    } else if (banishName.equals("staff of the standalone cheese")) {
-      String pref = monsterName;
-      String[] monsters = Preferences.getString("_jiggleCheesedMonsters").split("\\|");
-      for (int i = 0; i < monsters.length; ++i) {
-        if (monsters[i].length() > 0) {
-          pref += "|" + monsters[i];
+      case STAFF_OF_THE_STANDALONE_CHEESE:
+        {
+          Preferences.setString(
+              "_jiggleCheesedMonsters",
+              Stream.concat(
+                      Stream.of(monster.getName()),
+                      Arrays.stream(Preferences.getString("_jiggleCheesedMonsters").split("\\|"))
+                          .filter(Predicate.not(String::isEmpty)))
+                  .collect(Collectors.joining("|")));
+          break;
         }
-      }
-      Preferences.setString("_jiggleCheesedMonsters", pref);
     }
   }
 
-  private static void addBanishedMonster(
-      final String monsterName, final String banishName, final int turnBanished) {
-    BanishedMonster newBanishedMonster = new BanishedMonster(monsterName, banishName, turnBanished);
-    if (!BanishManager.banishedMonsters.contains(newBanishedMonster)) {
-      BanishManager.banishedMonsters.add(newBanishedMonster);
+  private static boolean addBanishedMonster(
+      final String monsterName, final Banisher banisher, final int turnBanished) {
+    var banishedMonster = new BanishedMonster(monsterName, banisher, turnBanished);
+    if (!banishedMonster.isValid()) {
+      return false;
     }
-  }
 
-  public static final void removeBanishByBanisher(final String banisher) {
+    banishedMonsters.add(banishedMonster);
+    return true;
+  }
 
-    BanishManager.banishedMonsters.removeIf(current -> current.getBanishName().equals(banisher));
+  public static final void removeBanishByBanisher(final String banisherName) {
```

Comment:
if this isn't used anywhere, you could delete it to keep the code using the type-safe version